From c3765421728e951d65cd92a9c4c9b2e356b694e2 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 6 Feb 2012 15:22:37 +0800
Subject: [PATCH 274/505] ENGR00173851 vpu: Fix decoder failure with dering
 enabled option

Return VIDIOC_QUERYBUF failed with "-e 1" option in vpu decoder.
This patch fixed it.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 101d034..b1d064a 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -1233,8 +1233,11 @@ decoder_start(struct decode *dec)
 					disp_clr_index = outinfo.indexFrameDisplay;
 					if (disp->buf.index != -1)
 						rotid = disp->buf.index; /* de-queued buffer as next rotation buffer */
-					else
+					else {
 						rotid++;
+						rotid = (rotid - dec->regfbcount) % dec->extrafb;
+						rotid += dec->regfbcount;
+					}
 				}
 				else
 					disp_clr_index = disp->buf.index;
-- 
1.8.0

